web/pages/group/[...slugs]/index.tsx

```diff
@@ -108,18 +109,27 @@ export async function getStaticPropz(props: { params: { slugs: string[] } }) {
   }
 }
 
-function toTopUsers(userScores: { [userId: string]: number }, users: User[]) {
+function toTopUsers(
+  userScores: { [userId: string]: number },
+  userIds: string[]
+) {
+  // sort userScores by score and trim to top 50
+  const sortedUserScores = Object.fromEntries(
+    Object.entries(userScores)
+      .sort(([, scoreA], [, scoreB]) => scoreB - scoreA)
+      .slice(0, 50)
+  ) as { [userId: string]: number }
+
   const topUserPairs = take(
-    sortBy(Object.entries(userScores), ([_, score]) => -1 * score),
+    sortBy(Object.entries(sortedUserScores), ([_, score]) => -1 * score),
     10
   ).filter(([_, score]) => score >= 0.5)
 
-  const topUsers = topUserPairs.map(
-    ([userId]) => users.filter((user) => user.id === userId)[0]
-  )
-  return topUsers.filter((user) => user)
+  const topUserIds = topUserPairs
+    .map(([userId]) => userIds.filter((uid) => uid === userId)[0])
+    .filter((userId) => userId != null) as string[]
```

Comment:
Could be:
`filterDefined(topUserPairs.map(([userId]) => userIds.find(userId)))`